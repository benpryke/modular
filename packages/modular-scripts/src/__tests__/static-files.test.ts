import execa from 'execa';
import * as fs from 'fs-extra';
import rimraf from 'rimraf';
import tree from 'tree-view-for-tests';
import path from 'path';
import getModularRoot from '../utils/getModularRoot';

const modularRoot = getModularRoot();

function modular(str: string, opts: Record<string, unknown> = {}) {
  return execa('yarnpkg', ['modular', ...str.split(' ')], {
    cwd: modularRoot,
    cleanup: true,
    ...opts,
  });
}

describe('WHEN working with a package with static directories', () => {
  beforeAll(async () => {
    rimraf.sync(path.join(modularRoot, 'dist', 'test-packages-static-files'));

    await modular('build test-packages-static-files', {
      stdio: 'inherit',
    });
  });

  it('THEN can build a package with static files', () => {
    expect(tree(path.join(modularRoot, 'dist', 'test-packages-static-files')))
      .toMatchInlineSnapshot(`
      "test-packages-static-files
      ├─ README.md #1jv3l2q
      ├─ dist-cjs
      │  ├─ test-packages-static-files.cjs.js #1y2lja9
      │  └─ test-packages-static-files.cjs.js.map #1nrziy
      ├─ dist-es
      │  ├─ test-packages-static-files.es.js #3hpj68
      │  └─ test-packages-static-files.es.js.map #dajrfy
      ├─ dist-types
      │  └─ src
      │     └─ index.d.ts #f68aj
      ├─ package.json
      └─ static
         └─ styles.css #h859ix"
    `);
  });

  it('THEN outputs the correct source map', async () => {
    expect(
      await fs.readJson(
        path.join(
          modularRoot,
          'dist',
          'test-packages-static-files',
          'dist-es',
          'test-packages-static-files.es.js.map',
        ),
      ),
    ).toMatchSnapshot();
  });

  it('THEN outputs the correct static file', async () => {
    expect(
      await fs.readFile(
        path.join(
          modularRoot,
          'dist',
          'test-packages-static-files',
          'static',
          'styles.css',
        ),
      ),
    ).toMatchSnapshot();
  });
});
