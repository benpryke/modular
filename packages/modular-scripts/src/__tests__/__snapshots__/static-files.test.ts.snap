// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WHEN working with a package with static directories THEN outputs the correct source map 1`] = `
Object {
  "file": "test-packages-static-files.es.js",
  "mappings": "AAAe,SAASA,GAAT,CAAaC,CAAb,EAAwBC,CAAxB,EAA2C;AACxD,SAAOD,CAAC,GAAGC,CAAX;AACD;;;;",
  "names": Array [
    "add",
    "a",
    "b",
  ],
  "sources": Array [
    "../src/index.ts",
  ],
  "sourcesContent": Array [
    "export default function add(a: number, b: number): number {
  return a + b;
}
",
  ],
  "version": 3,
}
`;

exports[`WHEN working with a package with static directories THEN outputs the correct static file 1`] = `
Object {
  "data": Array [
    109,
    97,
    105,
    110,
    32,
    123,
    10,
    32,
    32,
    109,
    97,
    114,
    103,
    105,
    110,
    58,
    32,
    53,
    59,
    10,
    125,
    10,
  ],
  "type": "Buffer",
}
`;
